lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1137,19 +1139,30 @@ static NodeValue simplifyConcatNode(Function *F, ConcatNode *CN) {
 }
 
 /// Optimize Concat nodes.
-static void optimizeConcatNodes(Function *F) {
-  auto &nodes = F->getNodes();
-
-  // For each node:
-  for (auto &node : nodes) {
-    if (auto *CN = dyn_cast<ConcatNode>(&node)) {
-      NodeValue newCN = simplifyConcatNode(F, CN);
-      if (newCN.getNode()) {
-        CN->getResult().replaceAllUsesOfWith(newCN);
+/// Optimize Concat nodes.
+class OptimizeConcatNodes : public Optimization {
+public:
+  OptimizeConcatNodes() : Optimization() {}
```

Comment:
if you are not planning to do anything here, just use `= default` syntax.
Same for .dtor